src/persist/src/indexed/unsealed.rs

```diff
@@ -358,33 +356,28 @@ impl Iterator for UnsealedSnapshotIter {
 
     fn next(&mut self) -> Option<Self::Item> {
         loop {
-            if self.current_batch.is_none() {
-                match self.batches.pop_front() {
-                    None => {
-                        return None;
-                    }
-                    Some(b) => {
-                        self.current_offset = 0;
-                        self.current_batch = Some(b.recv());
-                    }
+            if !self.current_batch.is_empty() {
+                let (kv, ts, diff) = self.current_batch.pop().unwrap();
+                if !(self.ts_lower.less_equal(&ts) && !self.ts_upper.less_equal(&ts)) {
```

Comment:
I'm having a hard time wrapping my head around these `not-less-equal` shenanigans. :sweat_smile: 

Is it correct that `ts_lower` is inclusive while `ts_upper` is exclusive? So we want an update if both `ts >= ts_lower` AND `ts < ts_upper`. Meaning, we want to reject an update if `!(ts_lower < ts)` OR `!(ts_upper <= ts)`? 

It's quite possible that I'm just too confused. :thinking:

Reply:
> I'm having a hard time wrapping my head around these not-less-equal shenanigans. 😅

You and me both!

> Is it correct that ts_lower is inclusive while ts_upper is exclusive?

Yup! This is documented on those two fields in the UnsealedSnapshotIter struct definition because I kept forgetting myself.

> Meaning, we want to reject an update if !(ts_lower < ts) OR !(ts_upper <= ts)? It's quite possible that I'm just too confused. 🤔

Yeah, theres some boolean logic theorem that `!(a AND b)` is rewriteable as `(!a OR !b)`. I've altered the control flow to return if it meets the "keep" condition instead of continuing if it doesn't, which removes a layer of boolean NOT and maybe makes this easier to follow? lmk what you think

Reply:
Oh boy! In the original version, I didn't see that it was !(a AND b). I thought it was !(a) AND !(b). The parentheses are important!

Reply:
The new version is more readable for me, though. :ok_hand: